collector/fixtures/e2e-output.txt

```diff
@@ -785,43 +788,82 @@ node_md_blocks_synced{device="md10"} 3.14159265e+08
 node_md_blocks_synced{device="md11"} 4.190208e+06
 node_md_blocks_synced{device="md12"} 3.886394368e+09
 node_md_blocks_synced{device="md127"} 3.12319552e+08
+node_md_blocks_synced{device="md13"} 4.88383936e+08
+node_md_blocks_synced{device="md20"} 9.76751616e+08
 node_md_blocks_synced{device="md219"} 7932
 node_md_blocks_synced{device="md3"} 5.853468288e+09
 node_md_blocks_synced{device="md4"} 4.883648e+06
+node_md_blocks_synced{device="md50"} 8.790400512e+09
 node_md_blocks_synced{device="md6"} 1.6775552e+07
 node_md_blocks_synced{device="md7"} 7.813735424e+09
 node_md_blocks_synced{device="md8"} 1.6775552e+07
 node_md_blocks_synced{device="md9"} 523968
-# HELP node_md_disks Total number of disks of device.
+# HELP node_md_disks Number of disks of device. (state="any" equals to the number of disks in the array)
```

Comment:
Looking at other players I've found this:
```viml
# HELP collectd_md_md_disks write_prometheus plugin: 'md' Type: 'md_disks', Dstype: 'gauge', Dsname: 'value'
# TYPE collectd_md_md_disks gauge
collectd_md_md_disks{md="0",type="active",instance="SynoDisk"} 3 1488700717769
collectd_md_md_disks{md="0",type="failed",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="0",type="missing",instance="SynoDisk"} 1 1488700717769
collectd_md_md_disks{md="0",type="spare",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="1",type="active",instance="SynoDisk"} 3 1488700717769
collectd_md_md_disks{md="1",type="failed",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="1",type="missing",instance="SynoDisk"} 1 1488700717769
collectd_md_md_disks{md="1",type="spare",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="2",type="active",instance="SynoDisk"} 1 1488700717769
collectd_md_md_disks{md="2",type="failed",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="2",type="missing",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="2",type="spare",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="3",type="active",instance="SynoDisk"} 2 1488700717769
collectd_md_md_disks{md="3",type="failed",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="3",type="missing",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="3",type="spare",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="5",type="active",instance="SynoDisk"} 3 1488700717770
collectd_md_md_disks{md="5",type="failed",instance="SynoDisk"} 0 1488700717770
collectd_md_md_disks{md="5",type="missing",instance="SynoDisk"} 0 1488700717770
collectd_md_md_disks{md="5",type="spare",instance="SynoDisk"} 0 1488700717770
collectd_md_md_disks{md="6",type="active",instance="SynoDisk"} 2 1488700717769
collectd_md_md_disks{md="6",type="failed",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="6",type="missing",instance="SynoDisk"} 0 1488700717769
collectd_md_md_disks{md="6",type="spare",instance="SynoDisk"} 0 1488700717769
```
The solution may be here: https://github.com/collectd/collectd/blob/master/src/md.c
Hoping it has the information for inactive raids as well. Looking into that...